opba-banking-protocol-facade/src/test/java/de/adorsys/opba/protocol/facade/services/authorization/ServiceSessionAndAuthSessionTest.java

```diff
@@ -0,0 +1,244 @@
+import lombok.SneakyThrows;
+import org.awaitility.Durations;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ActiveProfiles;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+
+/**
+ * Note: This test keeps DB in dirty state - doesn't cleanup after itself.
+ */
+@ActiveProfiles("test")
+@SpringBootTest(classes = ApplicationTest.class)
+public class ServiceSessionAndAuthSessionTest {
+    private static final String PASSWORD = "password";
+    private static final String TEST_BANK_ID = "53c47f54-b9a4-465a-8f77-bc6cd5f0cf46";
+    private static final UUID REQUEST_ID = UUID.fromString("e3865c6b-70f2-4c1e-ad31-d7c2ff160858");
+    private static final String REDIRECT_URL_OK = "http://google.com";
+    private static final String REDIRECT_URL_NO_OK = "http://microsoft.com";
+
+    @Autowired
+    private ListAccountsService listAccountsService;
+
+    @Autowired
+    private UpdateAuthorizationService updateAuthorizationService;
+
+    @Autowired
+    private ServiceSessionRepository serviceSessionRepository;
+
+    @Autowired
+    private AuthenticationSessionRepository authenticationSessions;
+
+    @MockBean
+    private Xs2aListAccountsEntrypoint xs2aListAccountsEntrypoint;
+
+    @MockBean
+    private Xs2aUpdateAuthorization xs2aUpdateAuthorization;
+
+    @Test
+    @SneakyThrows
+    void serviceSessionAndAuthSessionWithConsent_success() {
+        UUID sessionId = UUID.randomUUID();
+
+        // STEP 1
+        ValidationErrorResult validationErrorResult = buildValidationErrorResultResult();
+
+        doAnswer(invocation -> CompletableFuture.completedFuture(validationErrorResult))
+                .when(xs2aListAccountsEntrypoint)
+                .execute(any());
+
+        FacadeStartAuthorizationResult listAccountsResponse = (FacadeStartAuthorizationResult) listAccountsService.execute(buildListAccountRequest(sessionId)).get();
+
+        assertThat(UUID.fromString(listAccountsResponse.getAuthorizationSessionId())).isEqualTo(sessionId);
+        assertThat(UUID.fromString(listAccountsResponse.getServiceSessionId())).isEqualTo(sessionId);
+        assertThat(listAccountsResponse.getRedirectionTo()).isEqualTo(validationErrorResult.getRedirectionTo());
+
+        await().atMost(Durations.ONE_SECOND)
+                .pollDelay(Durations.ONE_HUNDRED_MILLISECONDS)
+                .until(() -> authenticationSessions.findAll().iterator().hasNext());
```

Comment:
I recommend to poll for 
`listAccountsResponse.getAuthorizationSessionId()`
so:
`authenticationSessions.findById(listAccountsResponse.getAuthorizationSessionId()).isPresent()`